src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2721,6 +2727,17 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 			for (int i = 1; i < fsig->param_count; ++i)
 				ins = emit_vector_insert_element (cfg, klass, ins, MONO_TYPE_R4, args [i + 1], i, FALSE);
 
+			if (len == 3) {
```

Comment:
Does this also handle the Vector3 constructor: `Vector3 (Vector2, float)`? As a follow up we could enable this block https://github.com/dotnet/runtime/blob/93381bf9c745e5925fb68ac4da2b38526b36222a/src/mono/mono/mini/simd-intrinsics.c#L2734-L2762 to intrinsify constructors that take another vector as a parameter.

Reply:
It is not supported yet, I created a new issue for the rest of vector constructors as it seems they don't work out of the box.

https://github.com/dotnet/runtime/issues/97815